cli/cmd/entitlements_set_value.go

```diff
@@ -0,0 +1,108 @@
+package cmd
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/replicatedhq/replicated/client/entitlements"
+	"net/url"
+	"os"
+	"strings"
+
+	"github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/log/level"
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+)
+
+func SetValueCmd() *cobra.Command {
+	vip := viper.New()
+
+	cmd := &cobra.Command{
+		Use:   "set-value",
+		Short: "API client for setting entitlements values",
+		Long: `
+`,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			stdoutLogger := log.NewLogfmtLogger(os.Stdout)
+			stdoutLogger = log.With(stdoutLogger, "ts", log.DefaultTimestampUTC)
+			if vip.GetBool("verbose") {
+				stdoutLogger = level.NewFilter(stdoutLogger, level.AllowDebug())
+			} else {
+				stdoutLogger = level.NewFilter(stdoutLogger, level.AllowWarn())
+			}
+
+			upstreamURL := vip.GetString("replicated-api-server")
+			upstream, err := url.Parse(upstreamURL)
+
+			if err != nil {
+				return errors.Wrapf(err, "parse replicated-api-server URL %s", upstreamURL)
+			}
+
+			token, err := require(vip, "replicated-api-token")
+			if err != nil {
+				return errors.Wrapf(err, "replicated-api-token or REPLICATED_API_TOKEN")
+			}
+
+			definitionsID, err := require(vip, "definitions-id")
+			if err != nil {
+				return err
+			}
+
+			customerID, err := require(vip, "customer-id")
+			if err != nil {
+				return err
+			}
+
+			key, err := require(vip, "key")
+			if err != nil {
+				return err
+			}
+
+			value, err := require(vip, "value")
+			if err != nil {
+				return err
+			}
+
+			datatype, err := require(vip, "type")
+			if err != nil {
+				return err
+			}
+
+			client := &entitlements.GraphQLClient{
+				GQLServer: upstream,
+				Token:     token,
+				Logger:    stdoutLogger,
+			}
+
+			created, err := client.SetEntitlementValue(customerID, definitionsID, key, value, datatype)
+
+			if err != nil {
+				return errors.Wrap(err, "set value")
+			}
+
+			bytes, _ := json.MarshalIndent(created, "", "  ")
+			fmt.Printf("%s\n", bytes)
+
+			return nil
+		},
+	}
+
+	cmd.Flags().String("replicated-api-token", "", "Token to use to communicate with https://g.replicated.com")
```

Comment:
why is this (and `replicated-app`, and `replicated-api-server`, and possibly others) a per-subcommand flag and not one for the parent command?

Reply:
ah good catch, that was a relic of when this was a standalone command tree. Will remove

Reply:
This turned into kind of a mess. Since this is somewhat temporary I'm considering just leaving this how it is.